File: go/vertical_slide/features/create_account/handler.go
package create_account

import (
	"ex/vertical_slide/shared/model"
	"ex/vertical_slide/shared/proto"
	"ex/vertical_slide/shared/repository"

	scyna "github.com/scyna/core"
)

const URL = "/account/create"

func Handler(ctx *scyna.Endpoint, request *proto.CreateAccountRequest) scyna.Error {
	ctx.Info("Receive CreateAccountRequest")

	if validateRequest(request) != nil {
		return scyna.REQUEST_INVALID
	}

	repository := repository.LoadRepository(&ctx.Context)

	if _, err := repository.GetAccount(request.Email); err == nil {
		return model.ACCOUNT_EXISTED
	}

	account := &model.Account{
		ID:       scyna.ID.Next(),
		Name:     request.Name,
		Email:    request.Email,
		Password: request.Password,
	}

	if err := repository.CreateAccount(account); err != nil {
		return err
	}

	return ctx.OK(&proto.CreateAccountResponse{Id: account.ID})
}

File: csharp/vertical_slice/features/GetAccount/Handler.cs
namespace ex.vsa.GetAccount;

using scyna;
using ex.vsa.model;
using ex.vsa.repository;

public class Handler : Endpoint.Handler<proto.RegisterAccountRequest>
{
    public override void Execute()
    {
        context.Info("Receive CreateAccountRequest");
        var repository = BaseRepository.load(context);
        var account = repository.GetAccountByEmail(request.Email);
        Response(new proto.GetAccountResponse
        {
            ID = account.ID,
            Name = account.Name,
            Email = account.Email,
        });
    }
}